src/core/init/utils/media_duration_updater.ts

```diff
@@ -139,7 +136,7 @@ export default class MediaDurationUpdater {
    * @param {number | undefined} newDuration
    */
   public updateKnownDuration(
-    newDuration : number | undefined
+    newDuration : IDurationItem | undefined
```

Comment:
After re-reading, I think asking for two arguments here instead of the `IDurationItem` object might be clearer in respect of what it will be doing to it, especially to the `isEnd` property. We could basically be more explicit here by reversing its value and naming it as a separate `addTimeMargin` boolean (as its effect is only to set a much higher value when actually setting the duration).